### DISCUSSION TITLE
Question about the IPM api: max_id_val_get and max_data_size_get

### DISCUSSION THREAD

**Discussion by Jingzhao123** [zephyrproject-rtos/zephyr#49428] [General]
**Question**
I need to create a IPM device, but I am not very clear about the IPM apis: max_id_val_get and max_data_size_get.
In latest ipm.h file, there is a comment about ipm_send API:

```
 * @param id Message identifier. Values are constrained by
 *        @a ipm_max_data_size_get since many boards only allow for a
 *        subset of bits in a 32-bit register to store the ID.
 * @param data Pointer to the data sent in the message.
 * @param size Size of the data.
 ........
__syscall int ipm_send(const struct device *ipmdev, int wait, uint32_t id,
		       const void *data, int size);
```

In my opinion, the param id in ipm_send function should be constrained by ipm_max_id_val_get, but not the ipm_max_data_size_get. The ipm_max_data_size_get should be used for constraining the data size, but not the max value of ID. 
Is my understanding correct? If not, could you please tell me how to use it and what does it mean?
Thanks.

**Comment by carlocaione** [comment]
We are trying to move away from IPM towards MBOX. Can you check if  the MBOX API solves your problem?

**Reply by khoa-nguyen-18** [reply]
Hi @carlocaione, could you please share us some information on the migration from IPM to MBOX?
